### PR TITLE
Mixed improvements part 2

### PR DESCRIPTION
**Pull request by ao-picterra**

### CODE REVIEW

**Comment by ao-picterra** [comment]
@colsil this is blocked (and on top of) #91 but you can start reviewing it against the platform master; I have a test file (in archive below) which helps testing a bit: it needs to be used inside the repo `examples/ ` folder in order to already have assets
`
[test.zip](https://github.com/Picterra/picterra-python/files/12049382/test.zip)

File: src/picterra/client.py [hunk 1]
```diff
@@ -725,5 +766,96 @@ def train_detector(self, detector_id):
...
         """
         resp = self.sess.post(self._api_url('detectors/%s/train/' % detector_id))
-        assert resp.status_code == 201, resp.status_code
+        if not resp.ok:
+            raise APIError(resp.text)
         self._wait_until_operation_completes(resp.json())
+
+
+    def run_advanced_tool(self, tool_id: UUID, inputs: dict, outputs: dict):
+            """
```
Comment (colsil) [inline src/picterra/client.py:775]: nit: is the indentation correct here? It looks like it's inconsistent with the other methods (8 spaces rather than 4?)

**Review by colsil — COMMENTED** [review]
One nit on indentation, other than that looks good to me.

File: src/picterra/client.py (hunk 1 above)
Comment (ao-picterra) [reply src/picterra/client.py]: Fixed

**Review by colsil — APPROVED** [review]

**Pull request merged**